.github/workflows/main-publish.yml

```diff
@@ -0,0 +1,31 @@
+# PyPi publish for Submit4DN
+
+name: publish 
+
+# Controls when the action will run. 
+on:
+
+  # Publish on all tags
+  push:
+    tags:
+    - '*'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-18.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - name: Publish
+        env:
+          PYPI_USER: ${{ secrets.PYPI_USER }}
+          PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: make publish
```

Comment:
I made this mistake for utils - make sure `poetry` is installed if that is the mechanism by which you are publishing.

Reply:
fixed I believe.